### PR TITLE
projects:iio_adpd1080:src: add clock calibarion routines

### PR DESCRIPTION
**Pull request by adrimbarean**
The ADPD1080 features two internal clocks that need to be calibrated: a
32kHz one and a 32MHz one. The 32kHz clock needs to be calibrated using an
external reference clock.

In this case a function is added to calibrate the
32kHz clock against an internal timer of the host controller.

Then a function to calibrate the 32MHz clock against the 32kHz clock is added.

Signed-off-by: Andrei Drimbarean <Andrei.Drimbarean@analog.com>

### CODE REVIEW

**Review by dbogdan — APPROVED** [review]

File: projects/iio_adpd1080/src/main.c [hunk 1]
```diff
@@ -47,13 +47,168 @@
...
+		goto finish;
+
+finish:
+	fin_ret = irq_ctrl_remove(cal_irq);
+	if(fin_ret != SUCCESS)
+		return FAILURE;
+gpio_finish:
+	fin_ret = gpio_remove(sync_gpio);
+	if(fin_ret != SUCCESS)
+		return FAILURE;
```
Comment (amiclaus) [inline projects/iio_adpd1080/src/main.c:203]: if you return here, `cal_timer` memory might remain allocated.

File: projects/iio_adpd1080/src/main.c [hunk 2]
```diff
@@ -47,13 +47,168 @@
...
+		goto finish;
+
+	status = adpd188_reg_write(adpd1080_dev, ADPD188_REG_GPIO_CTRL, 0);
+	if(status != SUCCESS)
+		goto finish;
+
+finish:
+	fin_ret = irq_ctrl_remove(cal_irq);
+	if(fin_ret != SUCCESS)
+		return FAILURE;
```
Comment (amiclaus) [inline projects/iio_adpd1080/src/main.c:199]: if you return here, `cal_timer` and `sync_gpio` memory might remain allocated.

Comment (scuciurean) [reply projects/iio_adpd1080/src/main.c:199]: If freeing the memory inside cal_irq's remove() function is failing, probably freeing the memory reserved for the gpio won't work either.

Comment (mchindri) [reply projects/iio_adpd1080/src/main.c:199]: I think it wouldn't be wrong to try :)

**Comment by adrimbarean** [comment]
V1:
 - remove returns at the end that might cause memory leaks.

**Review by amiclaus — APPROVED** [review]

**Pull request merged**